routing/http/client/transport.go

```diff
@@ -0,0 +1,38 @@
+package client
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+)
+
+type ResponseBodyLimitedTransport struct {
+	http.RoundTripper
+	LimitBytes int64
+}
+
+func (r *ResponseBodyLimitedTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp, err := r.RoundTripper.RoundTrip(req)
+	if resp != nil && resp.Body != nil {
+		resp.Body = &limitReadCloser{
+			limit:      r.LimitBytes,
+			ReadCloser: resp.Body,
+		}
+	}
+	return resp, err
+}
+
+type limitReadCloser struct {
+	limit     int64
+	bytesRead int64
+	io.ReadCloser
+}
+
+func (l *limitReadCloser) Read(p []byte) (int, error) {
+	n, err := l.ReadCloser.Read(p)
+	l.bytesRead += int64(n)
+	if l.bytesRead > l.limit {
```

Comment:
the https://pkg.go.dev/io#LimitReader has slightly more complex logic here - i don't know there's worry about determinism here in terms of having a successful read up to the limit, even if your reads on the underlying reader don't match up with the boundary

Reply:
Yeah I didn't use LimitReader because you can't tell if it stopped reading because it hit the limit, since it returns EOF in that case. So the error would be some JSON parsing error instead of "payload limit exceeded". 

The logic completely changes if you want to detect if you go *over* the limit, and it's non-trivial to do it precisely and I can't think of a practical reason that it needs to be precise. I'm guessing that, since you approved this, that you can't think of one that's worth pursuing either?